File: 1. Two Sum/Solution.java
//https://leetcode.com/problems/two-sum/

class Solution {
    public int[] twoSum(int[] nums, int target) {

    HashMap<Integer, Integer> n = new HashMap<>();
    for ( int i = 0 ; i <= nums.length ; i ++){
        if (n.containsKey(target-nums[i])){
            int number = n.get(target-nums[i]);
            return new int[] { number, i};
        }
        else{
            n.put(nums[i], i);
        }
    }
           

             return new int[] {-1,-1 };
	        
    }
}

File: 1. Two Sum/Solution.js
//https://leetcode.com/problems/two-sum/

/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (nums, target) {

    hash = {}
    for (var i = 0; i < nums.length; i++) {
        neededNumber = target - nums[i];
        if (neededNumber in hash) {
            return [hash[neededNumber], i]
        }
        else {
            hash[nums[i]] = i
        }
    }

    return [-1, -1]
};